stko/optimizers/collapser.py

```diff
@@ -228,60 +231,571 @@ def optimize(self, mol):
+            Target equilibrium bond length for long bonds to minimize
+            to.
+
+        num_steps : :class:`int`
+            Number of MC moves to perform.
+
+        bond_epsilon : :class:`float`, optional
+            Value of epsilon used in the bond potential in MC moves.
+            Determines strength of the bond potential.
+            Defaults to 50.
+
+        nonbond_epsilon : :class:`float`, optional
+            Value of epsilon used in the nonbond potential in MC moves.
+            Determines strength of the nonbond potential.
+            Defaults to 20.
+
+        nonbond_sigma : :class:`float`, optional
+            Value of sigma used in the nonbond potential in MC moves.
+            Defaults to 1.2.
+
+        nonbond_mu : :class:`float`, optional
+            Value of mu used in the nonbond potential in MC moves.
+            Determines the steepness of the nonbond potential.
+            Defaults to 3.
+
+        beta : :class:`float`, optional
+            Value of beta used in the in MC moves. Beta takes the
+            place of the inverse boltzmann temperature.
+            Defaults to 2.
+
+
+        """
+
+        self._output_dir = output_dir
+        self._step_size = step_size
+        self._target_bond_length = target_bond_length
+        self._num_steps = num_steps
+        self._bond_epsilon = bond_epsilon
+        self._nonbond_epsilon = nonbond_epsilon
+        self._nonbond_sigma = nonbond_sigma
+        self._nonbond_mu = nonbond_mu
+        self._beta = beta
+
+    def _get_bb_atom_ids(self, mol):
+
+        bb_ids = list(set([
+            i.get_building_block_id() for i in mol.get_atom_infos()
+        ]))
+        bb_atom_ids = {i: [] for i in bb_ids}
+        for i in mol.get_atom_infos():
+            bb_atom_ids[i.get_building_block_id()].append(
+                i.get_atom().get_id()
+            )
+
+        return bb_atom_ids
+
+    def _get_bond_length(self, mol, bond):
+
+        position_matrix = mol.get_position_matrix()
+        return get_atom_distance(
+            position_matrix=position_matrix,
+            atom1_id=bond.get_atom1().get_id(),
+            atom2_id=bond.get_atom2().get_id()
+        )
+
+    def _get_bond_vector(self, mol, bond):
+
+        position_matrix = mol.get_position_matrix()
+        atom1_pos = position_matrix[bond.get_atom1().get_id()]
+        atom2_pos = position_matrix[bond.get_atom2().get_id()]
+        return atom2_pos - atom1_pos
+
+    def _get_long_bond_infos(self, mol):
+        """
+        Returns dict of long bond infos.
+
+        """
+
+        long_bond_infos = {}
+        for bond_infos in mol.get_bond_infos():
+            if bond_infos.get_building_block() is None:
+                ids = (
+                    bond_infos.get_bond().get_atom1().get_id(),
+                    bond_infos.get_bond().get_atom2().get_id(),
+                )
+                long_bond_infos[ids] = bond_infos
+
+        return long_bond_infos
```

Comment:
Again, could be yield long_bond_infos one at a time.

Reply:
As above.